csharp/src/Ice/WSTransceiver.cs

```diff
@@ -1462,36 +1451,30 @@ private bool PostWrite(int size, ref int offset, int status)
 
         private bool ReadBuffered(int sz)
         {
-            if (_readBufferPos == _readBuffer.B.Position())
+            if (_readBufferPos == _readBufferOffset)
             {
-                _readBuffer.Resize(_readBufferSize, true);
+                _readBuffer = new byte[_readBufferSize];
                 _readBufferPos = 0;
-                _readBuffer.B.Position(0);
+                _readBufferOffset = 0;
             }
             else
             {
-                int available = _readBuffer.B.Position() - _readBufferPos;
+                int available = _readBufferOffset - _readBufferPos;
                 if (available < sz)
                 {
-                    if (_readBufferPos > 0)
-                    {
-                        _readBuffer.B.Limit(_readBuffer.B.Position());
-                        _readBuffer.B.Position(_readBufferPos);
-                        _readBuffer.B.Compact();
-                        Debug.Assert(_readBuffer.B.Position() == available);
-                    }
-                    _readBuffer.Resize(Math.Max(_readBufferSize, sz), true);
+                    byte[] tmpBuffer = new byte[Math.Max(_readBufferSize, sz)];
+                    _readBuffer.Slice(_readBufferPos, available).CopyTo(tmpBuffer);
+                    _readBuffer = tmpBuffer;
                     _readBufferPos = 0;
-                    _readBuffer.B.Position(available);
+                    _readBufferOffset = available;
```

Comment:
I'm pretty sure the previous code avoided any new memory allocation here, it just moved the data to the beginning of the read buffer. Why not just do the same here?

Reply:
The previous code call Compact to move data to the beginning, and then call Resize that will cause an allocation if required. I will update this to keep the same logic